src/jit/lower.cpp

```diff
@@ -788,6 +792,80 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
     isOnStack = info->regNum == REG_STK;
 #endif // !FEATURE_UNIX_AMD64_STRUCT_PASSING
 
+#ifdef _TARGET_ARM_
+    // Strcut can be splitted into register(s) and stack in ARM
+    if (info->isSplit)
+    {
+        if (arg->OperGet() == GT_FIELD_LIST)
+        {
+            GenTreeFieldList* fieldListPtr       = arg->AsFieldList();
+            GenTreeFieldList* regFieldListHeader = fieldListPtr;
+            GenTreeFieldList* stkFieldListHeader = nullptr;
+            GenTreeFieldList* prevPtr            = nullptr;
+            GenTreePtr        newOper            = nullptr;
+
+            assert(fieldListPtr->IsFieldListHead());
+
+            for (unsigned ctr = 0; fieldListPtr != nullptr; fieldListPtr = fieldListPtr->Rest(), ctr++)
```

Comment:
Please use a better name than `ctr` - in this case I actually think `fieldIndex` would be more meaningful.